buildSrc/src/test/java/com/osacky/flank/gradle/integration/SanityRoboTest.kt

```diff
@@ -0,0 +1,396 @@
+      |  roboScript = "some/path/script.json"
+      |}
+    """
+    )
+
+    val result = failedGradleRun(
+      arguments = listOf("printYml"),
+      projectDir = testProjectRoot.root
+    )
+
+    assertThat(result.output).contains("FAILED")
+    assertThat(result.output).contains(baseConfigMessage("roboScript"))
+  }
+
+  @Test
+  fun `sanityRobo - should throw an error if roboDirectives set`() {
+    makeBuildDotGradle(
+      where = testProjectRoot,
+      buildScript =
+        """
+      |plugins {
+      |  id "com.osacky.fladle"
+      |}
+      |
+      |fladle {
+      |  debugApk = "debug.apk"
+      |  sanityRobo = true
+      |  serviceAccountCredentials = layout.projectDirectory.file("flank-gradle-service.json")
+      |  roboDirectives = [
+      |    ["click", "button1", ""],
+      |    ["ignore", "button2"],
+      |    ["text", "field1", "my text"],
+      |  ]
+      |}
+    """
+    )
+
+    val result = failedGradleRun(
+      arguments = listOf("printYml"),
+      projectDir = testProjectRoot.root
+    )
+
+    assertThat(result.output).contains("FAILED")
+    assertThat(result.output).contains(baseConfigMessage("roboDirectives"))
+  }
+
+  @Test
+  fun `sanityRobo - should throw an error if additionalTestApks set`() {
+    makeBuildDotGradle(
+      where = testProjectRoot,
+      buildScript =
+        """
+      |plugins {
+      |  id "com.osacky.fladle"
+      |}
+      |
+      |fladle {
+      |  debugApk = "debug.apk"
+      |  sanityRobo = true
+      |  serviceAccountCredentials = layout.projectDirectory.file("flank-gradle-service.json")
+      |  additionalTestApks = [
+      |    "- app: debug2.apk",
+      |    "  test: test2.apk",
+      |    "- test: test3.apk"
+      |  ]
+      |}
+    """
+    )
+
+    val result = failedGradleRun(
+      arguments = listOf("printYml"),
+      projectDir = testProjectRoot.root
+    )
+
+    assertThat(result.output).contains("FAILED")
+    assertThat(result.output).contains(baseConfigMessage("additionalTestApks"))
+  }
+
+  @Test
+  fun `sanityRobo - should throw an error if roboScript set (multiple config)`() {
+    makeBuildDotGradle(
+      where = testProjectRoot,
+      buildScript =
+        """
+      |plugins {
+      |  id "com.osacky.fladle"
+      |}
+      |
+      |fladle {
+      |  serviceAccountCredentials = layout.projectDirectory.file("flank-gradle-service.json")
+      |  debugApk = "foo.apk"
+      |  instrumentationApk = "test.apk"
+      |  additionalTestApks = [
+      |    "- app: debug2.apk",
+      |    "  test: test2.apk",
+      |    "- test: test3.apk"
+      |  ]
+      |  configs {
+      |    sanity {
+      |      sanityRobo.set(true)
+      |      roboScript.set("path/to/script.json")
+      |    }
+      |  }
+      |}
+    """
+    )
+
+    val expectedMessage = additionalConfigMessage("roboScript", "sanity")
+
+    val result = failedGradleRun(
```

Comment:
shouldn't this one succeed? or am I missing something?

Reply:
My idea was to take the fail-fast approach and throw an error during the configuration phase

Reply:
Ah same thing. 👍